### PR TITLE
Flipper implementation

### PR DESCRIPTION
**Pull request by matthias-luger**

### CODE REVIEW

**Comment by Ekwav** [comment]
missing any sort of explanation

**Comment by Ekwav** [comment]
- Labels are wrong
- There is no indication that more recent flips become available when premium is purchased
- There is no indication that this is work in progress
- And that the default flips are a proof of concept
- "Latest profitable Auctions" should always be displayed with call to action buttons
     - Login
     - Purchase Premium (navigate to /premium)
- /premium has to list the flipper as a feature 
- There is no indication when a flip-title becomes clickable. There should be either a loading bar or circle.

File: src/api/ApiHelper.tsx [hunk 1]
```diff
@@ -254,18 +254,22 @@ function initAPI(): API {
...
+            window.localStorage.setItem("websocketUUID", websocketUUID);
+    
+            websocketHelper.sendRequest({
+                type: RequestType.SET_CONNECTION_ID,
+                data: websocketUUID,
+                resolve: () => {
+                    resolve();
+                 },
+                reject: (error: any) => {
+                    apiErrorHandler(RequestType.SET_CONNECTION_ID, error, websocketUUID);
```
Comment (Flou21) [inline src/api/ApiHelper.tsx:269]: reject?

File: src/api/WebsocketHelper.tsx [hunk 2]
```diff
@@ -43,6 +38,34 @@ function initWebsocket(): void {
         }
 
         removeSentRequests([...equals, request]);
+    }
+
+    let _handleSubscriptionOnMessage = function (response: ApiResponse, subscription: ApiSubscription) {
+        let parsedResponse = JSON.parse(response.data);
+        console.log(response)
+        console.log(response.type == "error")
```
Comment (Flou21) [inline src/api/WebsocketHelper.tsx:46]: why console.log

File: src/api/WebsocketHelper.tsx [hunk 3]
```diff
@@ -104,6 +127,27 @@ function sendRequest(request: ApiRequest): Promise<void> {
...
+            subscription.data = Base64.encode(subscription.data);
+        } catch (error) {
+            throw new Error("couldnt btoa this data: " + subscription.data);
+        }
+        apiSubscriptions.push(subscription);
+        websocket.send(JSON.stringify(subscription))
+
+    } else if (!websocket || websocket.readyState === WebSocket.CONNECTING) {
+        websocket.onopen = function () {
+            console.log("websocket opened");
```
Comment (Flou21) [inline src/api/WebsocketHelper.tsx:145]: why console log

File: src/components/Flipper/Flipper.tsx [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+            let updatedLastestAuctions = [newFipAuction, ...latestAuctions];
+            setLatestAuctions(updatedLastestAuctions);
+            latestAuctions = updatedLastestAuctions;
+
+            // reload for link-update
+            setTimeout(() => {
+                newFipAuction.showLink = true;
+                forceUpdate();
+            }, 5000)
+            console.log("hi")
```
Comment (Flou21) [inline src/components/Flipper/Flipper.tsx:39]: .

File: src/components/Flipper/Flipper.tsx [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+            <path d="M9.5 1a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5v-1a.5.5 0 0 1 .5-.5h3zm-3-1A1.5 1.5 0 0 0 5 1.5v1A1.5 1.5 0 0 0 6.5 4h3A1.5 1.5 0 0 0 11 2.5v-1A1.5 1.5 0 0 0 9.5 0h-3z" />
+        </svg>
+    );
+
+    let mapAuctionElements = auctions => {
+        return <div className="cards-wrapper">{
+            auctions.map((flipAuction) => {
+                return (
+                    <div className="card-wrapper" key={flipAuction.uuid}>
+                        <Card className="card" style={{ minHeight: "20vh" }}>
```
Comment (Flou21) [inline src/components/Flipper/Flipper.tsx:74]: inline style?

File: src/components/Flipper/Flipper.tsx [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+                )
+            })
+        }</div>;
+    };
+
+    return (
+        <div className="flipper">
+            <p>This flipper is work in progress (proof of concept/open alpha). Anything you see here is subject to change.
+                Please write us your opinion and suggestion on our <a href="https://discord.gg/Qm55WEkgu6">discord</a>.</p>
+            {latestAuctions.length > 0 ?
```
Comment (Flou21) [inline src/components/Flipper/Flipper.tsx:123]: default value when latestAuctions is 0?

File: src/api/ApiHelper.tsx (hunk 1 above)
Comment (Ekwav) [reply src/api/ApiHelper.tsx:269]: added reject

File: src/api/WebsocketHelper.tsx (hunk 2 above)
Comment (Ekwav) [reply src/api/WebsocketHelper.tsx:46]: forgot them

File: src/components/RecentAuctions/RecentAuctions.css [hunk 7]
```diff
@@ -9,19 +9,19 @@
     }
 }
 
-@media all and (min-width: 480px) and (max-width: 768px) {
+@media all and (min-width: 480px) and (max-width: 860px) {
```
Comment (Flou21) [inline src/components/RecentAuctions/RecentAuctions.css:12]: variables?

File: src/api/WebsocketHelper.tsx (hunk 3 above)
Comment (Ekwav) [reply src/api/WebsocketHelper.tsx:145]: removed

File: src/components/Flipper/Flipper.tsx (hunk 5 above)
Comment (Ekwav) [reply src/components/Flipper/Flipper.tsx:74]: removed

File: src/components/RecentAuctions/RecentAuctions.css (hunk 7 above)
Comment (Ekwav) [reply src/components/RecentAuctions/RecentAuctions.css:12]: not worth the overhead in this case

File: src/components/Flipper/Flipper.tsx [hunk 8]
```diff
@@ -0,0 +1,154 @@
...
+            <p>Volume is the amount of auctions that were sold in a 24 hour window.
+            It is capped at 60 to keep the flipper fast.</p>
+            <h3>I have another question</h3>
+            Ask via <a href="https://discord.gg/Qm55WEkgu6">discord</a> or <Link >feedback site</Link>
+
+        </div >
+    );
+}
+
+export default Flipper;
```
Comment (Flou21) [inline src/components/Flipper/Flipper.tsx:154]: hate the existence of that

**Review by Flou21 — APPROVED** [review]

**Pull request merged**